flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -428,6 +429,10 @@ public final void invoke() throws Exception {
 				// See FLINK-7430
 				isRunning = false;
 			}
+			MailboxExecutor mainMailboxExecutor = mailboxProcessor.getMainMailboxExecutor();
```

Comment:
There is an existing `AsyncWaitOperator` test that uses timers and blocks until it's elements are not resolved (by timeout). Without this change, the test would fail.